### DISCUSSION TITLE
Getting error: query is not a function, it is string.

### DISCUSSION THREAD

**Discussion by tGitm** [nextgenhealthcare/connect#4762] [Q&A] [answered]
This is my code in source transformer, I have another source transformer which is used for reading database (source is database reader for SQLite) and passing read data to outbound message. Now I need another on (code below), which will read another DB and pass missing data from another db to my outbound message.

When I run this channel I keep getting this error:
`[Source Filter/Transformer JavaScript Task on From DB to completed HL7 response (ba7dad39-ca73-42d2-8405-1537ab7a3835) < pool-1-thread-30] transformer: DB Exception: query is not a function, it is string.`

`var sqlDBName = 'jdbc:sqlite:D:/documents/MESI/mesi-integration/mesi-integration.db';
var sqlDBDriver = 'org.sqlite.JDBC';
var sqlDBUser = '';
var sqlDBPass = '';

try {
	var dbConn = DatabaseConnectionFactory.createDatabaseConnection(sqlDBDriver, sqlDBName, sqlDBUser, sqlDBPass);
	//query for getting data that we need in our HL7 message
	var query = 'SELECT '  +
		'Id,'+ 
		'VisitNumber,' + 
		'ActivityId,' + 
		'IntegrationServiceId,' +
		'TabletGroupId,' +
		'TaskType,' +
		'Status,' +
		'FirstName,' +
		'LastName,' +
		'BirthDate,' +
		'Gender,' +
		'Race,' +
		'EmailAddress,' +
		'PhoneNumber,' +
		'AddressStreet,' +
		'AddressPostalCode,' +
		'AddressCountry,' +
		'DoctorFirstName,' +
		'DoctorLastName,' +
		'TabletAcceptedBy,' +
		'RequestDate,' +
		'AddressCity ' +
		'FROM '+
		'Tasks' + ';';	
		
	var result = dbConn.executeCachedQuery(query);
	logger.info("DB Query: " + result);
	
	var id;
	var visitNumber;
	var activityId;
	var integrationId;
	var tabletGroupId;
	
	while (result.next()) {
		id = query('Id');
		visitNumber = query('VisitNumber');
		activityId = query('ActivityId');
		integrationId = query('IntegrationServiceId');
		tabletGroupId = query('TabletGroupId');
		logger.info(id);
		logger.info(visitNumber);
		logger.info(activityId);
		logger.info(integrationId);
		logger.info(tabletGroupId);
	}
	
	
	dbConn.close();
} catch(e) {
	logger.info("DB Exception: " + e.message);
}`

How to solve this?

**Comment by pacmano1** [comment, marked as answer]
your results are in "result", not the variable "query"

**Reply by pacmano1** [reply]
See https://forums.mirthproject.io/forum/mirth-connect/support/7963-retrieve-data-from-database?p=55478#post55478 as an example.  The forum has a lot of rich content that you should check out if you haven't.

**Reply by tGitm** [reply]
Thanks for explanation!

**Reply by tGitm** [reply]
Can you help me with this one please :)
https://github.com/nextgenhealthcare/connect/discussions/4760
